### PR TITLE
Add FilterObjects function 

### PR DESCRIPTION
**Pull request by omer9564**

### CODE REVIEW

File: pkg/tests/integration_test.go [hunk 1]
```diff
@@ -61,28 +95,35 @@ func checkBulk(ctx context.Context, t *testing.T, permitClient *permit.Client, r
...
-		Resource: enforcement.ResourceBuilder(resourceKey).WithTenant(tenantKey).Build(),
-		Context:  nil,
-	}
-	results, _ := permitClient.BulkCheck(requests...)
-	assert.Len(t, results, len(bulkAssignments)+1)
-	for i := 0; i <= len(bulkAssignments); i++ {
-		if i == len(bulkAssignments) {
-			assert.False(t, results[i])
-		} else {
+	//requests[len(bulkAssignments)] = enforcement.CheckRequest{
```
Comment (orishavit) [inline pkg/tests/integration_test.go:113]: delete

File: pkg/enforcement/bulk_check.go [hunk 2]
```diff
@@ -95,6 +95,48 @@ func (e *PermitEnforcer) parseBulkResponse(res *http.Response) ([]CheckResponse,
 }
 
 func (e *PermitEnforcer) BulkCheck(requests ...CheckRequest) ([]bool, error) {
+	tenantRequestsMap := make(map[string][]CheckRequest)
+	requestsOrder := make(map[string]map[int]int)
+
+	for i, request := range requests {
+		// Create a map of tenant key to requests for that tenant
+		tenantRequests := tenantRequestsMap[request.Resource.GetTenant()]
```
Comment (orishavit) [inline pkg/enforcement/bulk_check.go:103]: Use `tenantRequests, found := tenantRequestsMap[...]` syntax

**Review by orishavit — CHANGES_REQUESTED** [review]
A lot of these data structure manipulations can be made clearer with a library like `lo` (not urgent at all)

File: pkg/tests/integration_test.go (hunk 1 above)
Comment (omer9564) [reply pkg/tests/integration_test.go]: fixed

File: pkg/enforcement/bulk_check.go (hunk 2 above)
Comment (omer9564) [reply pkg/enforcement/bulk_check.go]: fixed

**Review by orishavit — APPROVED** [review]

File: pkg/enforcement/bulk_check.go (hunk 2 above)
Comment (omer9564) [reply pkg/enforcement/bulk_check.go]: fixed

File: pkg/tests/integration_test.go (hunk 1 above)
Comment (omer9564) [reply pkg/tests/integration_test.go]: fixed

**Pull request merged**